### DISCUSSION TITLE
INAV 8.0.0 RC1 Issue that Pops up on Target (gcc-arm-none-eabi 11.2.1 "is not implemented and will always fail")

### DISCUSSION THREAD

**Discussion by synersignart** [iNavFlight/inav#10502] [General]
Issue that Pops up on Target compilation

:/Workspace/xpack/xpack-arm-none-eabi-gcc-13.2.1-1.1/bin/../lib/gcc/arm-none-eabi/13.2.1/../../../../arm-none-eabi/bin/ld.exe: E:/Workspace/xpack/xpack-arm-none-eabi-gcc-13.2.1-1.1/bin/../lib/gcc/arm-none-eabi/13.2.1/../../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard\libc_nano.a(libc_a-closer.o): in function `_close_r':
closer.c:(.text._close_r+0xc): warning: _close is not implemented and will always fail

is this the correct Xpack for compilation of Target for this Version of INAV?
xpack-arm-none-eabi-gcc-13.2.1-1.1

![inav 8 0 0 target](https://github.com/user-attachments/assets/5cc247fa-d01f-4ab3-b7ef-439c08ee4aa5)
![inav 8 0 0-2 target](https://github.com/user-attachments/assets/f38ef453-9e21-42ee-93f3-0b1ba74a5360)

**Comment by mmosca** [comment]
Try using WSL instead of MSYS. I don't think anyone is activelly using MSYS these days.

https://github.com/iNavFlight/inav/blob/master/docs/development/Building%20in%20Windows%2010%20or%2011%20with%20Linux%20Subsystem.md

**Reply by synersignart** [reply]
> Try using WSL instead of MSYS. I don't think anyone is activelly using MSYS these days.
> 
> https://github.com/iNavFlight/inav/blob/master/docs/development/Building%20in%20Windows%2010%20or%2011%20with%20Linux%20Subsystem.md

Ive been using Msys since INAV5 and its always been reliable untill INAV 8.0.0  and Gcc xpack-arm-none-eabi-gcc-13.2.1
but not sure if something is missing base on the documentation folder if ever this was updated since INAV6.0.0

**Comment by stronnag** [comment]
The error is harmless. Linker optimisation will ignore the unneeded symbols.

When INAV installs a compiler, the install process removes the unwanted symbol section to avoid upsetting users with spurious **warnings**.

**Reply by synersignart** [reply]
ah I see  kinda surprise this pops up as it always been normal ever since INAV5 when I started using Msys

**Reply by MartinHugh** [reply]
Seems fine on WSL for me.

**Reply by stronnag** [reply]
> ah I see kinda surprise this pops up as it always been normal ever since INAV5 when I started using Msys

No surprise at all . It is expected from arm-gcc after about gcc 10 or 11.

**Comment by stronnag** [comment]
If it really bothers you, can "fix" it as we do (see `cmake/arm-none-eabi-checks.cmake`):
```
    # This is **somewhat ugly**
    # the newlib distributed by ARM generates suprious warnings from re-entrant POSIX functions
    # that INAV doesn't use. These "harmless" warnings can be surpressed by removing the
    # errant section from the only libnosys used by INAV ...
    # So look the other way ... while this is "fixed"
    execute_process(COMMAND arm-none-eabi-objcopy -w -R .gnu.warning.* "${TOOLS_DIR}/${base_dir_name}/arm-none-eabi/lib/thumb/v7e-m+fp/hard/libnosys.a"
      RESULT_VARIABLE status
      WORKING_DIRECTORY ${TOOLS_DIR}
    )

```
